timelock-server/src/main/java/com/palantir/atlasdb/timelock/clock/ClockServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.clock;
+
+public class ClockServiceImpl implements ClockService {
+    @Override
+    public long getSystemTimeInNanos() {
+        return System.nanoTime();
```

Comment:
I'm a little concerned about exposing & comparing this across distributed systems. [`System.currentTimeMillis()`](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#currentTimeMillis--) is more intended for this purpose; however, there are separate concerns about monotonicity.

See the JavaDoc for [System.html#nanoTime()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--) which states:

> This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time. The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative). The same origin is used by all invocations of this method in an instance of a Java virtual machine; other virtual machine instances are likely to use a different origin.

See also https://bugs.openjdk.java.net/browse/JDK-6458294 for some history.